hazelcast/src/main/java/com/hazelcast/instance/DefaultNodeExtension.java

```diff
@@ -234,7 +234,7 @@ public void shutdown() {
     @Override
     public void validateJoinRequest(JoinMessage joinMessage) {
         // check joining member's major.minor version is same as current cluster version's major.minor numbers
-        if (!joinMessage.getVersion().asClusterVersion().equals(node.getClusterService().getClusterVersion())) {
+        if (!joinMessage.getVersion().asVersion().equals(node.getClusterService().getClusterVersion())) {
             throw new VersionMismatchException("Joining node's version " + joinMessage.getVersion() + " is not compatible with "
```

Comment:
👍 just pushed another commit that renames `JoinMessage.version` to `memberVersion`